tests/helpers/test_entity_component.py

```diff
@@ -0,0 +1,236 @@
+"""
+# pylint: disable=protected-access,too-many-public-methods
+from collections import OrderedDict
+import logging
+import unittest
+from unittest.mock import patch, Mock
+
+import homeassistant.core as ha
+import homeassistant.loader as loader
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.entity_component import EntityComponent
+from homeassistant.components import discovery
+
+from tests.common import get_test_home_assistant, MockPlatform, MockModule
+
+_LOGGER = logging.getLogger(__name__)
+DOMAIN = "test_domain"
+
+
+class EntityTest(Entity):
+    def __init__(self, **values):
+        self._values = values
+
+        if 'entity_id' in values:
+            self.entity_id = values['entity_id']
+
+    @property
+    def name(self):
+        return self._handle('name')
+
+    @property
+    def should_poll(self):
+        return self._handle('should_poll')
+
+    @property
+    def unique_id(self):
+        return self._handle('unique_id')
+
+    def _handle(self, attr):
+        if attr in self._values:
+            return self._values[attr]
+        return getattr(super(), attr)
+
+
+class TestHelpersEntityComponent(unittest.TestCase):
+    """ Tests homeassistant.helpers.entity_component module. """
+
+    def setUp(self):  # pylint: disable=invalid-name
+        """Initialize a test Home Assistant instance."""
+        self.hass = get_test_home_assistant()
+
+    def tearDown(self):  # pylint: disable=invalid-name
+        """Clean up the test Home Assistant instance."""
+        self.hass.stop()
+
+    def test_setting_up_group(self):
+        component = EntityComponent(_LOGGER, DOMAIN, self.hass,
+                                    group_name='everyone')
+
+        # No group after setup
+        assert 0 == len(self.hass.states.entity_ids())
+
+        component.add_entities([EntityTest(name='hello')])
+
+        # group exists
+        assert 2 == len(self.hass.states.entity_ids())
+        assert ['group.everyone'] == self.hass.states.entity_ids('group')
+
+        group = self.hass.states.get('group.everyone')
+
+        assert ('test_domain.hello',) == group.attributes.get('entity_id')
+
+        # group extended
+        component.add_entities([EntityTest(name='hello2')])
+
+        assert 3 == len(self.hass.states.entity_ids())
+        group = self.hass.states.get('group.everyone')
+
+        assert ['test_domain.hello', 'test_domain.hello2'] == \
+            sorted(group.attributes.get('entity_id'))
+
+    @patch('homeassistant.helpers.entity_component.track_utc_time_change')
+    def test_polling_only_updates_entities_it_should_poll(self, mock_track):
+        component = EntityComponent(_LOGGER, DOMAIN, self.hass, 20)
+
+        no_poll_ent = EntityTest(should_poll=False)
+        no_poll_ent.update_ha_state = Mock()
+        poll_ent = EntityTest(should_poll=True)
+        poll_ent.update_ha_state = Mock()
+
+        component.add_entities([no_poll_ent])
+        assert not mock_track.called
```

Comment:
We're using py.test to run our tests. Py.test actually will break down your test if it is not correct. So it will say `mock_track.called` is actually this value. Applying `not` makes it this.